processor/elastictraceprocessor/go.mod

```diff
@@ -0,0 +1,73 @@
+module github.com/elastic/opentelemetry-collector-components/processor/elastictraceprocessor
+
+go 1.23.8
+
+require (
+	github.com/elastic/opentelemetry-lib v0.18.0
```

Comment:
I suggest purposely leaving this one on this version - even if there are newer releases to be on the safe side. Most of the logic of this processor is actually in this lib - the purpose to keep the `elastictrace` processor around is to have a smooth transition. For this, I'd not pull in changes that significantly extended the responsibility of `opentelemetry-lib` - I'd rather just keep it as is and then use the newer versions in the `elasticapm` processor.